pywr/recorders/recorders.py

```diff
@@ -110,7 +110,29 @@ def setup(self):
         if self.nodes is None:
             self.node_names = sorted(self.model.nodes.keys())
         else:
-            self.node_names = sorted(n.name for n in self.nodes)
+            # the next line is commented because when the nodes names are provided (e.g. loaded from the json)
+            # it will throw an exception
+            # self.node_names = sorted(n.name for n in self.nodes)
+            nodes = []
+            node_names = []
+            for node_ in self.nodes:
+                # test if the node name is provided
+                if isinstance(node_, basestring):
+                    try:
```

Comment:
We can probably drop this try-except, or at least it is inconsistent with `TablesRecorder` which just lets the `KeyError` from `NodeIterator` raise.

Reply:
I was trying to provide more information to the user when something is going wrong. But if you think the current approach is fine, I will drop try and catch

Reply:
Fair point, but I think you'll get the same information from the `KeyError` traceback i.e. it'll tell you whcih node was not found and that this happened in the `CSVRecorder`. Generally we haven't caught these errors internally in the other recorders.